controllers/storagecluster/platform_detection.go

```diff
@@ -54,13 +61,68 @@ func (p *Platform) getPlatform(c client.Client) (configv1.PlatformType, error) {
 	return p.platform, nil
 }
 
-func avoidObjectStore(p configv1.PlatformType) bool {
+func (r *StorageClusterReconciler) avoidObjectStore(instance *ocsv1.StorageCluster) (bool, configv1.PlatformType, error) {
```

Comment:
Don't return the PlatformType here. Since it is not always needed (you ignore it a lot, actually), just have whatever is calling this function also get the PlatformType itself.